### PR TITLE
[WIP] Add Eirini extensions and Eirini persi broker

### PR DESCRIPTION
**Pull request by mudler**
## Description

* Adds eirini-extensions and eirini-persi-broker jobs to the eirini pod
* Disables nfs-persi and enables eirini-persi-broker in postdeployment steps if eirini is enabled

## Test plan

```
$> make vagrant-prep
$> make run-eirini
$> git clone https://github.com/scf-samples/dizzylizard.git
$> cd dizzylizard
$> cf push --no-start
$> cf service-brokers
# You should see eirini-persi
$> cf enable-service-access eirini-persi
$> cf marketplace -s eirini-persi 
# Default plan should be available, based on the storageclass used for install

# Each service corresponds to a volume mount for the app, in this case mounting 2 volumes:
$> cf create-service eirini-persi default eirini-persi-1
$> cf create-service eirini-persi default eirini-persi-2
$> cf bind-service dizzylizard eirini-persi-1
$> cf bind-service dizzylizard eirini-persi-2

$> cf start dizzylizard 
# If you ssh to the app you would see in the env VCAP_SERVICES 
# the paths in the container which gives you access to the volumes

$> kubectl get pvc -n eirini
# Verifies the Persistent volume claims are bound (similarly for deletion)
```

### CODE REVIEW

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 1]
```diff
@@ -168,6 +168,38 @@ instance_groups:
...
+            min: 1
+            max: 1
+            ha: 1
+          memory: 256
+          virtual-cpus: 2
+        ports:
+        - name: broker
+          protocol: TCP
+          internal: 8999
+  - name: eirini-extensions
```
Comment (viovanov) [inline container-host-files/etc/scf/config/role-manifest.yml]: This needs `feature: eirini`

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 2]
```diff
@@ -168,6 +168,38 @@ instance_groups:
     release: scf
   - name: patch-properties
     release: scf-helper
+  - name: eirini-persi-broker
```
Comment (viovanov) [inline container-host-files/etc/scf/config/role-manifest.yml]: This needs feature: `eirini`

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 3]
```diff
@@ -3979,6 +4023,10 @@ variables:
   options:
     default: 'splatform/eirini-cert-copier:1.0.0.4.gd8e7208'
     description: "The docker image used by Eirini to register the image registry CA cert with Docker, on each Kubernetes node"
+- name: EIRINI_EIRINI_EXTENSIONS_SERVICE_HOST
+  options:
+    type: environment
+    description: "The Eirini extensions service ClusterIP"
```
Comment (viovanov) [inline container-host-files/etc/scf/config/role-manifest.yml]: Please mention that it's automatically set by Kubernetes

Comment (mudler) [reply container-host-files/etc/scf/config/role-manifest.yml]: Should be fixed in: https://github.com/SUSE/scf/pull/2397/commits/e2aca801407febacf5d0174fcc04aa672eb1ca47

File: container-host-files/etc/scf/config/role-manifest.yml (hunk 2 above)
Comment (mudler) [reply container-host-files/etc/scf/config/role-manifest.yml]: Should be covered by: https://github.com/SUSE/scf/blob/e2aca801407febacf5d0174fcc04aa672eb1ca47/container-host-files/etc/scf/config/role-manifest.yml#L157

File: container-host-files/etc/scf/config/role-manifest.yml [hunk 4]
```diff
@@ -2838,6 +2871,17 @@ configuration:
...
     properties.domain: '"((DOMAIN))"'
+    properties.eirini-extensions.kube_service_host: "((KUBERNETES_SERVICE_HOST))"
+    properties.eirini-extensions.kube_service_port: "((KUBERNETES_SERVICE_PORT))"
+    properties.eirini-extensions.namespace: "((EIRINI_KUBE_NAMESPACE))"
+    properties.eirini-extensions.operator_webhook_host: "((EIRINI_EIRINI_EXTENSIONS_SERVICE_HOST))"
+    properties.eirini-persi-broker.auth_password: '"((EIRINI_PERSI_NFS_BROKER_PASSWORD))"'
+    properties.eirini-persi-broker.kube_service_host: "((KUBERNETES_SERVICE_HOST))"
+    properties.eirini-persi-broker.kube_service_port: "((KUBERNETES_SERVICE_PORT))"
+    properties.eirini-persi-broker.namespace: "((EIRINI_KUBE_NAMESPACE))"
+    properties.eirini-persi-broker.service_plans: '[{"id" :"default-storageclass", "name":"default", "description":"Eirini persistence broker", "free": true, "kube_storage_class": "((KUBERNETES_STORAGE_CLASS_PERSISTENT))", "default_size": "1Gi" }]'
```
Comment (mudler) [inline container-host-files/etc/scf/config/role-manifest.yml]: ```KUBERNETES_STORAGE_CLASS_PERSISTENT``` is used only in tests, this needs to be adapted yet (don't merge it!)

**Pull request merged**